src/components/UserSelector.tsx

```diff
@@ -1,34 +1,26 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
-import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import * as usersSliceActions from '../features/users';
+import * as authorSliceActions from '../features/author';
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
-
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+export const UserSelector: React.FC = () => {
   const [expanded, setExpanded] = useState(false);
 
+  const { users } = useAppSelector(state => state.users);
+  const { author: selectedUser } = useAppSelector(state => state.author);
```

Comment:
Verify that the `state.author` structure matches the actual Redux state structure. If the state does not have an `author` property, this will cause an error.